pkg/cleaner/cleaner.go

```diff
@@ -0,0 +1,79 @@
+package cleaner
+
+import (
+	"context"
+	cassdcapi "github.com/datastax/cass-operator/operator/pkg/apis/cassandra/v1beta1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"log"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type Agent struct {
+	Client client.Client
+	Namespace string
+}
+
+func New(namespace string) (*Agent, error) {
+	_ = api.AddToScheme(scheme.Scheme)
+	_ = cassdcapi.AddToScheme(scheme.Scheme)
+
+	c, err := client.New(ctrl.GetConfigOrDie(), client.Options{ Scheme: scheme.Scheme })
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &Agent{
+		Client: c,
+		Namespace: namespace,
+	}, nil
+}
+
+func (a *Agent) RemoveResources() error {
+	// Remove CassandraDatacenter (cass-operator should delete all the finalizers and associated resources)
+	if err := a.removeCassandraDatacenter(); err != nil {
+		log.Fatalf("Failed to remove Cassandra cluster(s): %v", err)
+	}
+	return nil
+}
+
+func (a *Agent) removeCassandraDatacenter() error {
```

Comment:
This is going to delete all CassandraDatacenters in the namespace which could include ones that were not install by the Helm chart. We need to be careful and only delete ones installed by the chart. We probably need to filter release label.